client/dashboard/utils/url.ts

```diff
@@ -0,0 +1,7 @@
+export function isRelativeUrl( url: string ) {
+	try {
+		return new URL( url, window.location.href ).origin === window.location.origin;
```

Comment:
Will this work with, e.g., the backup URL `https://wordpress.com/backup/${ site.slug }`? The origin will be the same (wordpress.com), so it will try to treat it as internal link. But we do want to open in a new tab because it's not in HD; it's Calypso nav-unification.

I guess we should just `return url.startsWith('/')` instead? 🤔